xrpl4j-client/src/test/java/org/xrpl/xrpl4j/client/XrplClientTest.java

```diff
@@ -0,0 +1,1256 @@
+        Hash256 transactionHash
+      ) {
+        calledWithHash.set(transactionHash);
+        return Optional.empty();
+      }
+
+      @Override
+      protected UnsignedInteger getMostRecentlyValidatedLedgerIndex() {
+        return UnsignedInteger.valueOf(2);
+      }
+
+      @Override
+      public ServerInfo serverInfo() {
+        ServerInfoLedger serverInfoLedger = ServerInfoLedger.builder()
+          .hash(Hash256.of(Strings.repeat("0", 64)))
+          .age(UnsignedInteger.ONE)
+          .reserveBaseXrp(UnsignedInteger.ONE)
+          .reserveIncXrp(UnsignedInteger.ONE)
+          .sequence(LedgerIndex.of(UnsignedInteger.ONE))
+          .baseFeeXrp(BigDecimal.ONE)
+          .build();
+        ServerInfo serverInfo = ServerInfo.builder()
+          .completeLedgers("1-1")
+          .amendmentBlocked(true)
+          .buildVersion("1")
+          .closedLedger(serverInfoLedger)
+          .hostId("id")
+          .ioLatencyMs(UnsignedLong.valueOf(2))
+          .jqTransOverflow("flow")
+          .lastClose(ServerInfoLastClose.builder()
+            .convergeTimeSeconds(1.11)
+            .proposers(UnsignedInteger.ONE)
+            .build())
+          .publicKeyNode("node")
+          .serverState("full")
+          .serverStateDurationUs("10")
+          .time(ZonedDateTime.now())
+          .upTime(UnsignedLong.ONE)
+          .validationQuorum(UnsignedInteger.ONE)
+          .build();
+        return serverInfo;
+      }
+
+      @Override
+      public AccountInfoResult accountInfo(AccountInfoRequestParams params) {
+        AccountRootObject root = AccountRootObject.builder()
+          .accountTransactionId(Hash256.of(Strings.repeat("0", 64)))
+          .account(Address.of("rDgZZ3wyprx4ZqrGQUkquE9Fs2Xs8XBcdw"))
+          .balance(XrpCurrencyAmount.ofDrops(1000))
+          .ownerCount(UnsignedInteger.ONE)
+          .previousTransactionId(Hash256.of(Strings.repeat("0", 64)))
+          .previousTransactionLedgerSequence(UnsignedInteger.ONE)
+          .sequence(UnsignedInteger.ONE)
+          .flags(Flags.AccountRootFlags.DISABLE_MASTER)
+          .index(Hash256.of(Strings.repeat("0", 64)))
+          .build();
+        AccountInfoResult accountInfoResult = AccountInfoResult.builder()
+          .accountData(root)
+          .build();
+
+        return accountInfoResult;
+      }
+    };
+
+    Hash256 transactionHash = Hash256.of(Strings.repeat("0", 64));
+    assertThat(
+      xrplClient.isFinal(
+        transactionHash,
+        LedgerIndex.of(UnsignedInteger.ONE),
+        UnsignedInteger.ONE,
+        UnsignedInteger.ONE,
+        Address.of("rDgZZ3wyprx4ZqrGQUkquE9Fs2Xs8XBcdw")
+      ).finalityStatus()
+    ).isEqualTo(FinalityStatus.EXPIRED);
+
+    assertThat(calledWithHash.get()).isEqualTo(transactionHash);
+  }
+
+  @Test
+  void isFinalForNoLedgerGapAndLastLedgerSequencePassedTransaction_ReturnsExpired() {
```

Comment:
This test is slightly mis-named (it's not returning `EXPIRED`). How about this:

```suggestion
  void isFinalForNoLedgerGapAndLastLedgerSequencePassed_ReturnsExpiredWithSpentSequence() {
```